cdap-metadata-ext-spanner/src/main/java/io/cdap/cdap/metadata/spanner/SpannerConfig.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright © 2025 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.cdap.metadata.spanner;
+
+import com.google.cloud.spanner.DatabaseAdminClient;
+import com.google.cloud.spanner.Spanner;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Utility class for spanner metadata storage.
+ */
+class SpannerConfig {
```

Comment:
Introduce a parameterized constructor and pass `Map<String, String> cConf` as an arg.

Cache the values of project, instance and database in member variables.

Return these member variables in the getters.